### PR TITLE
Update displayed balance

### PR DESCRIPTION
**Pull request by Pinkdesu**
-Updates the balance value during operations
-Refactor the 'wallet' component

### CODE REVIEW

**Review by osipov-mit — APPROVED** [review]

**Review by EugenWay — APPROVED** [review]

File: website/frontend/src/components/blocks/Wallet/SelectAccountModal/SelectAccountModal.module.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.message {
+   text-align: center;
+ 
+   a {
+     text-decoration: underline;
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/SelectAccountModal/SelectAccountModal.module.scss:5]: I believe it's better to disable `text-decoration` only if `a` is outside of paragraph, in our `index.scss`
```
// underline only if link is a part of paragraph
  &:not(p &) {
    text-decoration: none;
  }
```

File: website/frontend/src/components/blocks/Wallet/SelectAccountModal/SelectAccountModal.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+const SelectAccountModal = (props: Props) => {
+  const alert = useAlert();
+  const { setAccount } = useAccount();
+  const injectedAccounts = useAccounts();
+
+  const { isOpen, onClose } = props;
+
+  // Setting current account and save it into the LocalStage
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/SelectAccountModal/SelectAccountModal.tsx:27]: remove this comment please

File: website/frontend/src/components/blocks/Wallet/Wallet.tsx [hunk 3]
```diff
@@ -1,28 +1,25 @@
...
+import { LOCAL_STORAGE } from 'consts';
+import { useAccount, useApi } from 'hooks';
 
-  const alert = useAlert();
+const Wallet = () => {
   const { api } = useApi();
+  const injectedAccounts = useAccounts();
   const { account: currentAccount, setAccount } = useAccount();
 
+  const [isOpen, setIsOpen] = useState(false);
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/Wallet.tsx:20]:
```suggestion
  const [isModalOpen, setIsModalOpen] = useState(false);
```

File: website/frontend/src/components/blocks/Wallet/Wallet.tsx [hunk 4]
```diff
@@ -1,28 +1,25 @@
...
-  const injectedAccounts = useAccounts();
-  const [accountBalance, setAccountBalance] = useState<string | null>(null);
-  const [isOpen, setIsOpen] = useState(false);
+import { LOCAL_STORAGE } from 'consts';
+import { useAccount, useApi } from 'hooks';
 
-  const alert = useAlert();
+const Wallet = () => {
   const { api } = useApi();
+  const injectedAccounts = useAccounts();
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/Wallet.tsx:17]:
```suggestion
  const accounts = useAccounts();
```
`useAccounts` executes `web3enable` polkadot api, therefore I would suggest to use it only once there, and then pass accounts as a prop to modal

File: website/frontend/src/components/blocks/Wallet/Wallet.tsx [hunk 5]
```diff
@@ -37,56 +34,38 @@ export const Wallet = () => {
...
+      if (unsub) {
+        unsub.then((callback) => callback());
       }
     };
-  }, [currentAccount]);
+  }, [api, currentAccount]);
 
-  const toggleModal = () => {
-    setIsOpen(!isOpen);
+  const handleModalOpen = () => {
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/Wallet.tsx:54]:
```suggestion
  const openModal = () => {
```

File: website/frontend/src/components/blocks/Wallet/Wallet.tsx [hunk 6]
```diff
@@ -37,56 +34,38 @@ export const Wallet = () => {
...
-        if (i === index) {
-          localStorage.setItem(LOCAL_STORAGE.SAVED_ACCOUNT, acc.address);
-          localStorage.setItem(LOCAL_STORAGE.PUBLIC_KEY_RAW, GearKeyring.decodeAddress(acc.address));
-        }
-      });
-      setAccount(injectedAccounts[index]);
-      toggleModal();
-      alert.success('Account successfully changed');
-    }
+  const handleModalClose = () => {
```
Comment (nikitayutanov) [inline website/frontend/src/components/blocks/Wallet/Wallet.tsx:58]:
```suggestion
  const closeModal = () => {
```

File: website/frontend/src/tests/Header.test.tsx [hunk 7]
```diff
@@ -265,8 +265,19 @@ describe('header tests', () => {
...
-      useApiSpy.mockImplementation(() => ({ api: { balance: { findOut: async () => ({ toHuman: () => value }) } } }));
+      useApiSpy.mockImplementation(() => ({
+        api: {
+          balance: { findOut: async () => ({ toHuman: () => value }) },
+          //it works strangely, you need to study the issue more deeply
+          gearEvents: {
+            subscribeToBalanceChange: subscribeToBalanceChangeMock.mockResolvedValue(unsubMock),
+          },
+        },
+      }));
```
Comment (nikitayutanov) [inline website/frontend/src/tests/Header.test.tsx:280]: yeah, need to think about it

**Review by nikitayutanov — APPROVED** [review]

**Pull request merged**